modules/Bio/EnsEMBL/Compara/RunnableDB/MLSSIDFactory.pm

```diff
@@ -71,10 +71,36 @@ sub write_output {
     foreach my $method (keys %$methods) {
         my $branch_number = $methods->{$method};
         my $mlsss = $mlss_a->fetch_all_by_method_link_type($method);
-        foreach my $mlss (@$mlsss) {
-            $self->dataflow_output_id( { 'mlss_id' => $mlss->dbID }, $branch_number);
+
+        if ( $self->param('line_count') ) {
+            foreach my $mlss ( @$mlsss ) {
+                my $line_count = $self->_get_line_count($mlss);
+                $self->dataflow_output_id( { 'mlss_id' => $mlss->dbID, 'exp_line_count' => $line_count }, $branch_number);
+            }
+        }
+        else {
+            foreach my $mlss ( @$mlsss ) {
+                $self->dataflow_output_id( { 'mlss_id' => $mlss->dbID }, $branch_number);
+            }
         }
     }
 }
 
+sub _get_line_count {
+    my ($self, $mlss) = @_;
+    my $adaptor;
+
+    if ($mlss->method->class =~ /homology/i) {
+        $adaptor = $self->compara_dba->get_HomologyAdaptor;
+    }
+    elsif ($mlss->method->class =~ /synteny/i) {
+        $adaptor = $self->compara_dba->get_SyntenyRegionAdaptor;
+    }
+    else {
+        return;
+    }
+
+    my $allitems = $adaptor->fetch_all_by_MethodLinkSpeciesSet($mlss);
+    return scalar(@$allitems);
```

Comment:
Fetching the objects will be "slow" because it needs to transfer the data (rows) from the server to the Runnable and then instantiate all the objects. If you can count the objects directly in SQL and that would be much faster.

There is a standard way of doing that (introduced for `copy_data.pl`). Some adaptors implement a method named `count_by_mlss_id`. The standard way of implement this method is like in `SyntenyRegionAdaptor` or `ConstrainedElementAdaptor` or `GenomicAlignAdaptor`: you can use the parent class' `generic_count` method, which will read from the table that is set in `_tables`. If you implement the method in HomologyAdaptor, you can use it here, and any other method_link that needs a line_count would be added by implementing the method in its adaptor.